packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -173,30 +180,86 @@ export class ApolloServer extends ApolloServerBase {
         }
       }
 
-      const callbackFilter: APIGatewayProxyCallback = (error, result) => {
-        callback(
-          error,
-          result && {
-            ...result,
-            headers: {
-              ...result.headers,
-              ...requestCorsHeaders,
+      const makeCallbackFilter = (doneFn: any): APIGatewayProxyCallback => {
+        return (error, result) => {
+          doneFn();
+          callback(
+            error,
+            result && {
+              ...result,
+              headers: {
+                ...result.headers,
+                ...requestCorsHeaders,
+              },
             },
-          },
-        );
+          );
+        };
       };
 
-      graphqlLambda(async () => {
-        // In a world where this `createHandler` was async, we might avoid this
-        // but since we don't want to introduce a breaking change to this API
-        // (by switching it to `async`), we'll leverage the
-        // `GraphQLServerOptions`, which are dynamically built on each request,
-        // to `await` the `promiseWillStart` which we kicked off at the top of
-        // this method to ensure that it runs to completion (which is part of
-        // its contract) prior to processing the request.
-        await promiseWillStart;
-        return this.createGraphQLServerOptions(event, context);
-      })(event, context, callbackFilter);
+      const response = new Stream.Writable();
+
+      fileUploadProcess(event, response, this.uploadsConfig || {})
+        .then((body: any) => {
+          event.body = body;
+          graphqlLambda(async () => {
+            // In a world where this `createHandler` was async, we might avoid this
+            // but since we don't want to introduce a breaking change to this API
+            // (by switching it to `async`), we'll leverage the
+            // `GraphQLServerOptions`, which are dynamically built on each request,
+            // to `await` the `promiseWillStart` which we kicked off at the top of
+            // this method to ensure that it runs to completion (which is part of
+            // its contract) prior to processing the request.
+            await promiseWillStart;
```

Comment:
The `promiseWillStart` must be awaited prior to entering the `fileUploadProcess`.  We can't wait until after processing the file upload to fire this callback.